deptool.py

```diff
@@ -0,0 +1,249 @@
+
+            for s in trans.newsolvables():
+               print ','.join(packages), s.name
+               if opts.explain and s.name in opts.explain:
+                   reason, rule = solver.describe_decision(s)
+                   ruleinfo = None
+                   if rule:
+                       ruleinfo = rule.info().problemstr()
+                   print("-> %s %s %s"%(s.name, REASONS[reason], ruleinfo))
+
+            if opts.size:
+                size = trans.calc_installsizechange()
+                print("SIZE %s"%(size))
+
+            return True
+
+        if opts.single:
+            for n in args:
+                sel = self.pool.select(str(n), solv.Selection.SELECTION_NAME)
+                for s in sel.solvables():
+                    deps = s.lookup_deparray(solv.SOLVABLE_RECOMMENDS)
+                    deps += s.lookup_deparray(solv.SOLVABLE_REQUIRES)
+                    for dep in deps:
+                        # only add recommends that exist as packages
+                        rec = self.pool.select(dep.str(), solv.Selection.SELECTION_NAME)
+                        if not rec.isempty():
+                            if not solveit([dep.str()]):
+                                good = False
+        else:
+            if not solveit(args):
+                good = False
+
+        if not good:
+            logger.error("solver errors encountered")
+            return 1
+
+    @cmdln.option("-r", "--repo", dest="repo", action="append",
+                  help="repo to use")
+    def do_deps(self, subcmd, opts, *packages):
+        """${cmd_name}: show package deps
+
+        ${cmd_usage}
+        ${cmd_option_list}
+        """
+
+        self.prepare_pool(opts.repo)
+
+        for n in packages:
+            sel = self.pool.select(n, solv.Selection.SELECTION_NAME)
+            if sel.isempty():
+                logger.error("%s not found", n)
+            for s in sel.solvables():
+                print('- {}-{}@{}:'.format(s.name, s.evr, s.arch))
+                for kind in ('RECOMMENDS', 'REQUIRES', 'SUPPLEMENTS', 'ENHANCES', 'PROVIDES', 'SUGGESTS'):
+                    deps = s.lookup_deparray(getattr(solv, 'SOLVABLE_'+kind, 0))
+                    if deps:
+                        print('  {}:'.format(kind))
+                        for dep in deps:
+                            print('    - {}'.format(dep))
+
+    @cmdln.option("-r", "--repo", dest="repo", action="append",
+                  help="repo to use")
+    def do_whatprovides(self, subcmd, opts, *relation):
+        """${cmd_name}: list packages providing given relations
+
+        ${cmd_usage}
+        ${cmd_option_list}
+        """
+
+        self.prepare_pool(opts.repo)
+
+        for r in relation:
+            i = self.pool.str2id(r)
+            for s in self.pool.whatprovides(i):
+                print('- {}-{}@{}:'.format(s.name, s.evr, s.arch))
+
+if __name__ == "__main__":
+    app = DepTool()
+    sys.exit(app.main())
+
+# vim: sw=4 et
```

Comment:
:(

Reply:
it's my boilerplate. sorry for that :) didn't bother cleaning it up too much before showing the code to dimstar